libnetwork/ipams/defaultipam/allocator.go

```diff
@@ -64,6 +83,10 @@ func NewAllocator(lcAs, glAs []*ipamutils.NetworkToSplit) (*Allocator, error) {
 		return nil, fmt.Errorf("could not construct local address space: %w", err)
 	}
 
+	if len(lcAs6) == 0 {
+		lcAs6 = []*ipamutils.NetworkToSplit{deriveULABaseNetwork(hostID)}
+	}
```

Comment:
Is defaultipam the right place for this logic to live? The Host ID is a property of the daemon, and the address spaces are daemon config. Does the allocator really need to have knowledge of a host ID? Why don't IPAM plugins get feature parity?

I think that the daemon package should be responsible for setting the policy on when an automatically derived ULA prefix is to be used for IPv6 subnet allocations. Neither the allocator nor libnetwork need any modifications: the daemon could simply augment the user-supplied list (when appropriate) and leverage the existing `nwconfig.OptionDefaultAddressPoolConfig` to pass the resulting set of networks into libnetwork and ultimately the allocator.